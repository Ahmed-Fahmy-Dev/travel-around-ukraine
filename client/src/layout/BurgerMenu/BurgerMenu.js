import React, { useState } from 'react';
import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import Menu from '@mui/material/Menu';
import MenuIcon from '@mui/icons-material/Menu';
import MenuItem from '@mui/material/MenuItem';
import { NavLink } from 'react-router-dom';
import styles from './BurgerMenu.module.scss';

const pages = [
  {
    name: 'Home',
    link: '/',
    innerContent: 'Home',
  },
  {
    name: 'Catalogue',
    link: '/catalogue',
    innerContent: 'Catalogue',
  },
  {
    name: 'Cart',
    link: '/cart',
    innerContent: 'Cart',
  },
];

const BurgerMenu = () => {
  const [anchorElNav, setAnchorElNav] = useState(null);

  const handleOpenNavMenu = (event) => {
    setAnchorElNav(event.currentTarget);
  };

  const handleCloseNavMenu = () => {
    setAnchorElNav(null);
  };

  const boxSX = {
    flexGrow: 1,
    display: { sx: 'flex', laptop: 'none' },
    justifyContent: 'flex-end',
  };

  const menuIconSX = {
    '&:hover': {
      color: 'yellow',
      transition: '0.5s',
    },
  };

  const menuSX = {
    display: { sx: 'flex', laptop: 'none' },
  };

  const menuItemSX = {
    padding: 0,
    width: '40vw',
    maxWidth: '200px',
    height: '50px',
  };

  const anchorOrigin = {
    vertical: 'bottom',
    horizontal: 'left',
  };

  const transformOrigin = {
    vertical: 'top',
    horizontal: 'left',
  };

  return (
    <Box sx={boxSX}>
      <IconButton
        size="large"
        aria-label="appbar"
        aria-controls="menu-appbar"
        aria-haspopup="true"
        onClick={handleOpenNavMenu}
        color="inherit"
        sx={{ padding: 0 }}
      >
        <MenuIcon sx={menuIconSX} />
      </IconButton>

      <Menu
        id="menu-appbar"
        anchorEl={anchorElNav}
        anchorOrigin={anchorOrigin}
        keepMounted
        transformOrigin={transformOrigin}
        open={!!anchorElNav}
        onClose={handleCloseNavMenu}
        sx={menuSX}
      >
        {pages.map(({ name, link, innerContent }) => (
          <MenuItem key={name} onClick={handleCloseNavMenu} sx={menuItemSX}>
            <NavLink to={link} key={name} onClick={handleCloseNavMenu} className={styles.menuItemLink}>
              {innerContent}
            </NavLink>
          </MenuItem>
        ))}
      </Menu>
    </Box>
  );
};

export default BurgerMenu;
