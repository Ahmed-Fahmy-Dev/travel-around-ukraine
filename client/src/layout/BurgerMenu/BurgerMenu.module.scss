.menuItemLink{
    display: block;
    text-decoration: none;
    color: black;
    width: 100%;
    padding: 6px 16px;
}