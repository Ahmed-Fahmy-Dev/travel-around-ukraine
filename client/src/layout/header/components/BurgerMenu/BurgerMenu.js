import React, { useState } from 'react';
import { Box, MenuItem, Menu, IconButton, Typography } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';

const pages = [
  {
    name: 'home',
    link: '/',
    innerContent: 'Home',
  },
  {
    name: 'catalogue',
    link: '/catalogue',
    innerContent: 'Catalogue',
  },
  {
    name: 'logIn',
    link: '/login',
    innerContent: 'Log In',
  },
  {
    name: 'account',
    link: '/user',
    innerContent: 'Account',
  },
  {
    name: 'cart',
    link: '/cart',
    innerContent: 'Cart',
  },
  {
    name: 'favorites',
    link: '/favorites',
    innerContent: 'Favorites',
  },
];

const styleNavLink = {
  display: 'block',
  textDecoration: 'none',
  color: 'text.primary',
  width: '100%',
  padding: '6px 16px',
};

const boxSX = {
  flexGrow: 1,
  display: { xs: 'flex', laptop: 'none' },
  justifyContent: 'flex-end',
};

const menuSX = {
  display: { xs: 'flex', laptop: 'none' },
};

const menuItemSX = {
  padding: 0,
  width: '40vw',
  maxWidth: '200px',
  height: '50px',
};

const anchorOrigin = {
  vertical: 'bottom',
  horizontal: 'left',
};

const transformOrigin = {
  vertical: 'top',
  horizontal: 'left',
};

const BurgerMenu = () => {
  const [anchorElNav, setAnchorElNav] = useState(null);
  const { isLogin } = useSelector((store) => store.userReducer);

  const handleOpenNavMenu = (event) => {
    setAnchorElNav(event.currentTarget);
  };

  const handleCloseNavMenu = () => {
    setAnchorElNav(null);
  };

  return (
    <Box sx={boxSX}>
      <IconButton
        size="large"
        aria-label="appbar"
        aria-controls="menu-appbar"
        aria-haspopup="true"
        onClick={handleOpenNavMenu}
        sx={{ padding: 0, color: 'text.primary' }}
      >
        <MenuIcon />
      </IconButton>

      <Menu
        id="menu-appbar"
        anchorEl={anchorElNav}
        anchorOrigin={anchorOrigin}
        keepMounted
        transformOrigin={transformOrigin}
        open={!!anchorElNav}
        onClose={handleCloseNavMenu}
        sx={menuSX}
      >
        {pages.map(({ name, link, innerContent }) => {
          if (name === 'logIn' && isLogin) {
            return null;
          }
          if (name === 'account' && !isLogin) {
            return null;
          }

          return (
            <MenuItem key={name} onClick={handleCloseNavMenu} sx={menuItemSX}>
              <Typography component={NavLink} to={link} key={name} sx={styleNavLink} end>
                {innerContent}
              </Typography>
            </MenuItem>
          );
        })}
      </Menu>
    </Box>
  );
};

export default BurgerMenu;
