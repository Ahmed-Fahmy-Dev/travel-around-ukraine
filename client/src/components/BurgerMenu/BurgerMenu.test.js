import { render } from '@testing-library/react';
import BurgerMenu from './BurgerMenu';
import { BrowserRouter } from 'react-router-dom';

describe('Render BurgerMenu', () => {
  test('should BurgerMenu render', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <BurgerMenu />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
