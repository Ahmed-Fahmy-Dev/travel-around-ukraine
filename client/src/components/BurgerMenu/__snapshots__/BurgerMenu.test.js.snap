// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BurgerMenu should BurgerMenu render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-12lz99j"
  >
    <button
      aria-controls="menu-appbar"
      aria-haspopup="true"
      aria-label="appbar"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-gmtkzn-MuiButtonBase-root-MuiIconButton-root"
      data-testid="burgerBtn"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
